src/Metadata/PropertyTypeArray.php

```diff
@@ -6,6 +6,9 @@
 
 use Doctrine\Common\Collections\Collection;
 
+/**
+ * @todo rename to PropertyTypeIterable
```

Comment:
inspired by symfony workflow, i would recommend that we already create that new class and have it extend from PropertyTypeArray and start using the new class internally.

then the new class can already have the BC-breaking constructor as well. and you can mark this class as `@deprecated` and in version 2 we simply move the code and delete the deprecated class

Reply:
the problem with that is that it is not strictly BC. the metadata parser would start building a different class than before and if a consumer does `instanceof` it breaks. i think we need to un-final the PropertyTypeArray (its in the same repository, so we can reliably do that)